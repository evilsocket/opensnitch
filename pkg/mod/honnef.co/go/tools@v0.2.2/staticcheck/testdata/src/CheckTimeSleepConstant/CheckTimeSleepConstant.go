package pkg

import "time"

const c1 = 1
const c2 = 200

func fn() {
	time.Sleep(1)  // want `sleeping for 1`
	time.Sleep(42) // want `sleeping for 42`
	time.Sleep(201)
	time.Sleep(c1)
	time.Sleep(c2)
	time.Sleep(2 * time.Nanosecond)
	time.Sleep(time.Nanosecond)
}
