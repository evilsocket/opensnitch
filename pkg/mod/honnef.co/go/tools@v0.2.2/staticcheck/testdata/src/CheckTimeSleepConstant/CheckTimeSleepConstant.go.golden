-- explicitly use nanoseconds --
package pkg

import "time"

const c1 = 1
const c2 = 200

func fn() {
	time.Sleep(1 * time.Nanosecond)  // want `sleeping for 1`
	time.Sleep(42 * time.Nanosecond) // want `sleeping for 42`
	time.Sleep(201)
	time.Sleep(c1)
	time.Sleep(c2)
	time.Sleep(2 * time.Nanosecond)
	time.Sleep(time.Nanosecond)
}

-- use seconds --
package pkg

import "time"

const c1 = 1
const c2 = 200

func fn() {
	time.Sleep(1 * time.Second)  // want `sleeping for 1`
	time.Sleep(42 * time.Second) // want `sleeping for 42`
	time.Sleep(201)
	time.Sleep(c1)
	time.Sleep(c2)
	time.Sleep(2 * time.Nanosecond)
	time.Sleep(time.Nanosecond)
}
